cpp/include/experimental/graph_functions.hpp

```diff
@@ -243,5 +243,49 @@ void relabel(raft::handle_t const& handle,
              vertex_t num_labels,
              bool do_expensive_check = false);
 
+/**
```

Comment:
So, the current guideline (for myself) is to place graph analytics (e.g. PageRank, BFS, ...) in the existing algorithms.hpp while providing operations on graphs (but that does not modify the graph object, ones modifying the graph object needs to be a member function) in the graph_functions.hpp.

Do you have better suggestions for header file namings and where to put which?

Reply:
I can see how there's a thin and somewhat subjective boundary between graph analytics algos and operations on a graph that do not modify it. We should identify if there's a strong benefit to C++ API users (since it is an exposed header) and go from there.